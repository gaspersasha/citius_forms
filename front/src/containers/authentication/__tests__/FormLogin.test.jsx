import React from 'react';
import { FormLogin } from '..';

const props = {
  extendedActiveForm: jest.fn(),
  setEmail: jest.fn(),
  setPassword: jest.fn(),
  setRememberUser: jest.fn(),
  setFieldValue: jest.fn(),
  onSubmit: jest.fn(),
  isUserRemembered: true,
  errorMessage: '',
  emailValue: 'test@email.com',
};

describe('<FormLogin />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FormLogin {...props} />);
  });

  it('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call extendedActiveForm function with signup', () => {
    const signUpButton = wrapper.find('b[data-id="link"]');

    signUpButton.simulate('click');
    expect(props.extendedActiveForm).toHaveBeenCalledWith('signup');
  });

  it('should call extendedActiveForm function with reset', () => {
    const resetButton = wrapper.find('span[role="button"]');

    resetButton.simulate('click');
    expect(props.extendedActiveForm).toHaveBeenCalledWith('reset');
  });
});
