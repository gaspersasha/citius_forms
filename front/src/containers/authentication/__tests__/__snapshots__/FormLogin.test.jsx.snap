// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormLogin /> should match snapshots 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="wrapper"
  >
    <AuthForm
      activeForm="login"
      emailValue="test@email.com"
      errorMessage=""
      firstNameValue=""
      isUserRemembered={true}
      lastNameValue=""
      onSubmit={[MockFunction]}
      setFieldValue={[MockFunction]}
      subTitle="Enter your email address and password"
      title="Sign in to your BuyaCar account"
    />
    <div
      className="row row__middleBottomSpace"
    >
      <span
        className="typo typo__small forgotPassword"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        Forgot password?
      </span>
    </div>
    <p
      className="typo typo__small link"
    >
      <span>
        Donâ€™t have an account? 
      </span>
      <b
        data-id="link"
        onClick={[Function]}
        role="dialog"
      >
        Sign up
      </b>
    </p>
  </div>
</form>
`;
